File: app/src/main/java/com/example/digitalmonk/data/local/db/dao/UsageLogDao.java
package com.example.digitalmonk.data.local.db.dao;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import com.example.digitalmonk.data.local.db.entity.UsageLogEntity;
import java.util.List;

@Dao
public interface UsageLogDao {
    @Insert
    void insert(UsageLogEntity log);

    @Query("SELECT * FROM usage_logs ORDER BY timestamp DESC")
    List<UsageLogEntity> getAllLogs();

    @Query("DELETE FROM usage_logs")
    void deleteAll();
}

File: app/src/main/java/com/curbme/app/data/local/db/dao/UsageLogDao.kt
package com.curbme.app.data.local.db.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import com.curbme.app.data.local.db.entity.UsageLogEntity

@Dao
interface UsageLogDao {
    @Insert
    fun insert(log: UsageLogEntity)

    @Query("SELECT * FROM usage_logs ORDER BY timestamp DESC")
    fun getAllLogs(): List<UsageLogEntity>

    @Query("DELETE FROM usage_logs")
    fun deleteAll()
}